lib/ProofOfPossessionBuilder.ts

```diff
@@ -0,0 +1,26 @@
+import { createProofOfPossession } from './functions';
+import { PROOF_CANT_BE_CONSTRUCTED, ProofOfPossession, ProofOfPossessionOpts } from './types';
+
+export class ProofOfPossessionBuilder {
+  popCallbackOpts?: ProofOfPossessionOpts;
+  popEncoded?: ProofOfPossession;
+
+  withPoPCallbackOpts(popCallbackOpts: ProofOfPossessionOpts): ProofOfPossessionBuilder {
```

Comment:
```suggestion
  withProofCallbackOpts(proofCallbackOpts: ProofOfPossessionOpts): ProofOfPossessionBuilder {
```

This is already the Proof of Possession builder, so it is logical that the proof is about possession. Sugest to rename it to the above (of course you need to change the param names in the code as well